File: bubble_sort/python/bubble_sort.py
#!/usr/bin/env python3

from typing import List

def bubble_sort(arr: List, size: int) -> None:
  for i in range(size):
    for j in range(1, size - i):
      if arr[j - 1] > arr[j]:
        arr[j - 1], arr[j] = arr[j], arr[j - 1]

if __name__ == "__main__":
  arr = [6, 3, 5, 10, -1, 7, -4]
  bubble_sort(arr, len(arr))

File: algorithms/bubble_sort/c/bubble_sort.c
#include <stdio.h>

void bubble_sort(int *arr, size_t sz)
{
  size_t i, j;
  int tmp;

  for (i = 0; i < sz; i++) {
    for (j = 1; j < sz - i; j++) {
      if (arr[j - 1] > arr[j]) {
        tmp = arr[j - 1];
        arr[j - 1] = arr[j];
        arr[j] = tmp;
      }
    }
  }
}

int main(void)
{
  int arr[] = {5, -1, 3, 8, -4, 0};
  size_t sz = 6;

  bubble_sort(arr, sz);
  return 0;
}
